WSO2 EI creatio connector initial implementation (#1)
wso2-extensions/esb-connector-creatio: README.md

# Creatio REST EI Connector

> The Creatio REST Connector allows you to work with below CRM functinalities

Reviewer: ```suggestion
The Creatio REST Connector allows you to work with below CRM functionalities.
```

3. The Creatio connector zip file is created in the `esb-connector-creatiocrm/target` directory

Follow the steps given below to build the Class mediation whihc is dependecy library:

1. Get a clone or download the source from [Github](https://github.com/wso2-extensions/esb-connector-creatio/).
2. Run the following Maven command from the `CookieMediator` directory: `mvn clean install`.
3. Copy the CookieMediator-1.0.0-SNAPSHOT.jar into <EI-HOME>/micro-integrator/dropins directory

#### Deploy connector with mediator implemented CAR file 

1. Add connector creatiocrm-connector-1.1.0.zip to Intergration Studio inorder to work with EI tooling, see [Working with Connectors via Tooling](https://docs.wso2.com/display/EI650/Working+with+Connectors+via+Tooling) and create the mediation

2. Import the creatio server certificate to the EI client keystore using either the following command or the EI Management Console.

```
keytool -importcert -file <certificate_file> -keystore <EI>/repository/resources/security/client-truststore.jks -alias "creatio"

```

3. Deploy the car app to <EI-HOME>/micro-integrator/repository/deployment/server/carbonapps folder in WSO2 EI

> 4. Execute the below command encrypt the above config values ./ciphertool.sh -Dconfigure, provide the passowrd of wso2carbon.jks, default value is 'wso2carbon' if encrypted values need to be configured.

Reviewer: ```suggestion
4. Execute the below command to encrypt the above config values ./ciphertool.sh -Dconfigure, provide the password of wso2carbon.jks, default value is 'wso2carbon' if encrypted values need to be configured.
```

> 5. Start the WSO2 EI instanace like this. (you need to provide the passowrd of wso2carbon.jks at the execution, default value is 'wso2carbon' if values are encrypted )

Reviewer: ```suggestion
5. Start the WSO2 EI instance like this. (you need to provide the password of wso2carbon.jks at the execution, default value is 'wso2carbon' if values are encrypted )
```

	./micro-integrator.sh
6. Send the sample request to created mediation using CreatioCRM connector.

## Headers

Headers : 
Content-Type: application/json
Content-Length :

## Contact Operations

> 

Reviewer: please remove these unnecessary lines

Author: Removed

1. Create Contact

Creatio Contact create operation used in EI mediation. It will used payload to create contact in creatio

<creatiocrm.contactCreate/>
   
   

2. Get Contact

Ex:
    <creatiocrm.contactGet>
        <id>{$ctx:uri.var.id}</id>

>     </creatiocrm.contactGet>

Reviewer: we can use `xml tags to format these

Author: Corrected

    
3. Patch Contact

<creatiocrm.contactUpdate>
<id>{$ctx:uri.var.id}</id>
</creatiocrm.contactUpdate>

4. Delete Contact

<creatiocrm.contactDelete>
<id>{$ctx:uri.var.id}</id>
</creatiocrm.contactDelete>

## Case Operations

Body Format: 

> {

Reviewer: `json tages here

Author: Formated
